packages/superset-ui-legacy-plugin-chart-table/src/ReactDataTable.tsx

```diff
@@ -203,29 +202,39 @@ export default function ReactDataTable(props: DataTableProps) {
 
   return (
     <div className="superset-legacy-chart-table" ref={rootElem}>
-      <table className="table table-striped table-condensed table-hover dataTable">
+      <table className="table table-striped table-condensed table-hover">
         <thead>
           <tr>
             {columns.map(col => (
-              <th key={col.key}>{col.label}</th>
+              // by default all columns will have sorting
+              <th className="sorting" key={col.key} title={col.label}>
+                {col.label}
+              </th>
             ))}
           </tr>
         </thead>
         <tbody>
           {data.map((record, i) => (
-            <tr key={i}>
+            // hide rows after first page makes the initial render faster (less layout computation)
+            <tr key={i} style={{ display: pageLength > 0 && i >= pageLength ? 'none' : undefined }}>
```

Comment:
Even with pagination, do we still create all rows in dom?

Reply:
`jquery.datatables` either reads DOM attributes to collect data or we pass a `data` object with all the records. But the latter approach is actually slower because it also creates DOM nodes for all rows beforehand and calls "innerHTML" on every cell which we have no control of.